sklearn/naive_bayes.py

```diff
@@ -137,16 +143,26 @@ class GaussianNB(BaseNB):
     >>> from sklearn.naive_bayes import GaussianNB
     >>> clf = GaussianNB()
     >>> clf.fit(X, Y)
-    GaussianNB()
+    GaussianNB(class_prior_override=None)
     >>> print(clf.predict([[-0.8, -1]]))
     [1]
     >>> clf_pf = GaussianNB()
     >>> clf_pf.partial_fit(X, Y, np.unique(Y))
-    GaussianNB()
+    GaussianNB(class_prior_override=None)
     >>> print(clf_pf.predict([[-0.8, -1]]))
     [1]
     """
 
+    def __init__(self, class_prior_override=None):
+        # Check if we need to override the class priors
```

Comment:
These input checks should not be done at class instantiation. Here it should be just

    self.class_prior = class_prior